final/background/spoofax.tex

```diff
@@ -1,57 +1,89 @@
 \section{The Spoofax Language Workbench}
 \label{sec:spoofax}
-
-Spoofax is a platform that allows for giving a completely
-\emph{declarative} definition of a programming language and accompanying
-IDE support~\cite{Kats10a}. Such a platform is called a \emph{language
-workbench}. The definition of a programming language is done using
-high-level \emph{meta-languages} for each aspect of the programming
-language.
-
-To define a language declaratively means that one uses the
-meta-languages to specify \emph{what} the properties of a language are, not
-\emph{how} these properties are implemented. For example, instead of asking
-\emph{​``How do I implement a tokenizer and parser for my language?''​}, one
-asks \emph{​``What is the syntax of my language?''​}. From such a description
-in a meta-language, the tokenizer and parser can be derived without
-the designer of the language ever having to care about its
-implementation.
-
-This section goes over the aspects that come into play with the
-development of a language and how Spoofax tackles each of these
-aspects. First, the section goes over the elements that make up the
-specification of a language\footnote{This section follows
-the structure of the language specification portion of the compiler
-construction course at the TU Delft. The slides can be found here:
-\url{http://tudelft-in4303.github.io/lectures/specification/}.}. A language
-specification consists of the following conceptual steps:
+Spoofax is a platform that allows for giving a completely \emph{declarative}
+definition of a programming language and accompanying IDE
+support~\cite{Kats10a}. Such a platform is called a \emph{language workbench}.
+
+In discussing programming languages, it is useful to distinguish the following
+three concepts: an \textit{aspect} of a language, the formal
+\textit{specification} of that aspect, and the actual \textit{implementation} of
+that aspect, which follows from the specification. Any programming language can
+be divided into multiple distinct aspects. In \cref{fig:spoofax-relations}, four
+common aspects of programming languages are shown, along with the relation
+between the formal specification and implementation for each aspect. An example
+of an aspect is the \textit{syntax definition}: its formal specification is done
+with grammar rules, and its implementation consists of both a tokenizer and
+parser.
+
+A key property of Spoofax is that it allows a language designer to give a
+complete specification of each aspect of their language, and automatically
+derive the implementation of that language aspect from its
```

Comment:
`and automatically derives the implementations of those aspects from their specification`?